client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -420,6 +424,20 @@ private int executorCores(SparkConf conf) {
     }
   }
 
+  private void checkUserClassPathFirst() {
+    if ((Boolean) conf.get(package$.MODULE$.EXECUTOR_USER_CLASS_PATH_FIRST())) {
```

Comment:
@cfmcgrady, this has already respected the config `celeborn.client.spark.shuffle.fallback.policy` because `getReader/getWriter` is invoked after `registerShuffle`. Meanwhile, `registerShuffle` could not distinguish the problem case via classloader. Therefore, we could not perform the check within `CelebornShuffleFallbackPolicyRunner`. PTAL.

Reply:
@cfmcgrady, I have updated the description of the case that highly recommend to set `spark.executor.userClassPathFirst` to false. PTAL.